app/src/main/java/org/dhis2/data/forms/dataentry/fields/optionset/OptionSetSelectionView.java

```diff
@@ -171,10 +171,10 @@ private void setCheckOptions() {
                 optionBinding.setOptionName(option.displayName());
                 optionBinding.setVariable(BR.isBgTransparent, isBgTransparent);
                 optionBinding.checkBox.setLayoutParams(new LinearLayout.LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT, 1.0f));
-                optionBinding.checkBox.setChecked(currentCodeValue != null && (currentCodeValue.equals(option.code()) || currentCodeValue.equals(option.name())));
+                optionBinding.checkBox.setChecked(currentCodeValue != null && (currentCodeValue.equals(option.displayName()) || currentCodeValue.equals(option.name())));
```

Comment:
I see in **setOnChangeListener** that we are sending the _code_ which i think is the best way due to is always the same and non dependant from translations or typos.
So why in **setChecked** we are checking by _displayName_ or _name_ and not for _code_?